docs/interactions/Slash_Commands.md

```diff
@@ -0,0 +1,841 @@
+                            "description": "The channel permissions to edit",
+                            "type": 7,
+                            "required": false
+                        }
+                    ]
+                }
+            ]
+        },
+        {
+            "name": "role",
+            "description": "Get or edit permissions for a role",
+            "type": 2,
+            "options": [
+                {
+                    "name": "get",
+                    "description": "Get permissions for a role",
+                    "type": 1,
+                    "options": [
+                        {
+                            "name": "role",
+                            "description": "The role to get",
+                            "type": 8, // 8 is type ROLE
+                            "required": true
+                        },
+                        {
+                            "name": "channel",
+                            "description": "The channel permissions to get",
+                            "type": 7,
+                            "required": false
+                        }
+                    ]
+                },
+                {
+                    "name": "edit",
+                    "description": "Edit permissions for a role",
+                    "type": 1,
+                    "options": [
+                        {
+                            "name": "role",
+                            "description": "The role to edit",
+                            "type": 8,
+                            "required": true
+                        },
+                        {
+                            "name": "channel",
+                            "description": "The channel permissions to edit",
+                            "type": 7,
+                            "required": false
+                        }
+                    ]
+                }
+            ]
+        }
+    ]
+}
+```
+
+And, done! The JSON looks a bit complicated, but what we've ended up with is a single command that can be scoped to multiple actions, and then further scoped to a particular resource, and then even _further_ scope with optional arguments. Here's what it looks like all put together.
+
+![A command with grouped subcommands and parameters. It says /permissions user get with arguments for a user and a channel.](command-with-groups-subcommands-parameters.png)
+
+## Endpoints
+
+> info
+> For authoriation, all endpoints take either a bot token or client credentials token for your application
+
+## Get Global Application Commands % GET /applications/{application.id#DOCS_TOPICS_OAUTH2/application-object}/commands
+
+Fetch all of the global commands for your application. Returns an array of [ApplicationCommand](#DOCS_INTERACTIONS_SLASH_COMMANDS/applicationcommand) objects.
+
+## Create Global Application Command % POST /applications/{application.id#DOCS_TOPICS_OAUTH2/application-object}/commands
```

Comment:
Is there a limit to the payload size for this endpoint? (Hopefully it's large enough, the previous permission example was 1kb minified 😰)

Reply:
A limit to the payload size for creating commands?

Reply:
Like how there's limits to embeds and whatnot, I guess. I was mostly just asking for the sake of asking, though.

Reply:
Those limits are mostly just because our `send_message` has a max filesize on it I believe